sharktuner/dispatch_tuner/dispatch_tuner.py

```diff
@@ -0,0 +1,149 @@
+# Copyright 2024 Advanced Micro Devices, Inc
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import logging
+import argparse
+import shutil
+from pathlib import Path
+from sharktuner import libtuner
+from sharktuner import common
+
+import pyarrow as pa
+import pyarrow.parquet as pq
+from datetime import datetime
+from dataclasses import dataclass, asdict, fields
+import hashlib
+import os
+from typing import Any
+
+
+class DispatchTuner(libtuner.TuningClient):
+    def __init__(self, tuner_context: common.TunerContext):
+        super().__init__(tuner_context)
+        self.compile_flags: list[str] = []
+        self.benchmark_flags: list[str] = []
+        self.compile_timeout: int = 16
+        self.benchmark_timeout: int = 16
+
+    def get_iree_compile_flags(self) -> list[str]:
```

Comment:
Are there some `@override` annotations we could add here? https://peps.python.org/pep-0698/